composer.json

```diff
@@ -13,21 +13,20 @@
     "prefer-stable": true,
     "minimum-stability": "beta",
     "require": {
-        "php": ">=5.5",
+        "php": "^5.5 || ^7.0",
         "ext-curl": "*",
         "php-http/httplug": "^1.0",
-        "php-http/message-factory": "^1.0"
+        "php-http/message-factory": "^1.0.2",
+        "php-http/message": "^1.2"
     },
     "suggest": {
         "php-http/discovery": "Allow automatically discover needed HTTPlug implementations"
     },
     "require-dev": {
         "guzzlehttp/psr7": "^1.0",
-        "php-http/adapter-integration-tests": "0.4.*",
-        "php-http/message": "^1.2",
-        "php-http/discovery": "~0.8.0",
+        "php-http/client-integration-tests": "^0.5.1",
```

Comment:
The ^ operator respects semver even for versions under 1.0. 

^0.5.1 is equal to >=0.5.1 <6.0.0.